### PR TITLE
Get rid of SidebarItem.props.subMenus in favour of props.children

### PR DESCRIPTION
**Pull request by ixrock**
Follow-up for https://github.com/lensapp/lens/pull/2366#pullrequestreview-619665907

### CODE REVIEW

File: src/renderer/components/layout/sidebar-item.tsx [hunk 1]
```diff
@@ -53,11 +52,9 @@ export class SidebarItem extends React.Component<SidebarItemProps> {
   }
 
   @computed get isExpandable(): boolean {
-    if (this.compact) {
-      return false; // not available currently
-    }
+    if (this.compact) return false; // not available in compact-mode currently
 
-    return Boolean(this.props.subMenus || this.props.children);
+    return Boolean(this.props.children);
```
Comment (Nokel81) [inline src/renderer/components/layout/sidebar-item.tsx:57]: how about:

```ts
    return !this.compact || Boolean(this.props.children);
```

File: src/renderer/components/layout/sidebar.tsx [hunk 2]
```diff
@@ -51,25 +51,22 @@ export class Sidebar extends React.Component<Props> {
...
+          {crds.map((crd) => {
+            return (
+              <SidebarItem
+                key={crd.getResourceApiBase()}
+                id={`crd-resource:${crd.getResourceApiBase()}`}
+                url={crd.getResourceUrl()}
+                text={crd.getResourceTitle()}
+              />
+            );
+          })}
```
Comment (Nokel81) [inline src/renderer/components/layout/sidebar.tsx]:
```suggestion
          {crds.map(crd => (
            <SidebarItem
              key={crd.getResourceApiBase()}
              id={`crd-resource:${crd.getResourceApiBase()}`}
              url={crd.getResourceUrl()}
              text={crd.getResourceTitle()}
            />
          ))}
```

**Review by Nokel81 — COMMENTED** [review]
Thanks

File: src/renderer/components/layout/sidebar-item.tsx (hunk 1 above)
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx:57]: I would like keep it as is for reasons:
1. current code imho more readable / intention clear
2. we could (and should) support it for `compact`-mode, so the line will be just removed.

Comment (Nokel81) [reply src/renderer/components/layout/sidebar-item.tsx:57]: okay

Comment (Nokel81) [reply src/renderer/components/layout/sidebar-item.tsx:57]: Though why did you format it at all then?

File: src/renderer/components/layout/sidebar.tsx (hunk 2 above)
Comment (ixrock) [reply src/renderer/components/layout/sidebar.tsx]: Updated

**Review by Nokel81 — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]

**Pull request merged**